src/hooks/useIsEncrypted.ts

```diff
@@ -6,19 +6,28 @@ SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
 Please see LICENSE files in the repository root for full details.
 */
 
-import { useCallback, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { MatrixClient, MatrixEvent, Room, RoomStateEvent, EventType } from "matrix-js-sdk/src/matrix";
 
 import { useTypedEventEmitter } from "./useEventEmitter";
 
-// Hook to simplify watching whether a Matrix room is encrypted, returns undefined if room is undefined
-export function useIsEncrypted(cli: MatrixClient, room?: Room): boolean | undefined {
-    const [isEncrypted, setIsEncrypted] = useState(room ? cli.isRoomEncrypted(room.roomId) : undefined);
+// Hook to simplify watching whether a Matrix room is encrypted, returns undefined if room is undefined, return null when isRoomEncrypted is computed
+export function useIsEncrypted(cli: MatrixClient, room?: Room): boolean | undefined | null {
+    const [isEncrypted, setIsEncrypted] = useState<boolean | null | undefined>(null);
 
+    useEffect(() => {
+        const func = async (): Promise<void> =>
+            room && setIsEncrypted(await cli.getCrypto()?.isEncryptionEnabledInRoom(room.roomId));
+        func();
+    }, [cli, room]);
     const update = useCallback(
-        (event: MatrixEvent) => {
+        async (event: MatrixEvent) => {
             if (room && event.getType() === EventType.RoomEncryption) {
-                setIsEncrypted(cli.isRoomEncrypted(room.roomId));
+                try {
+                    setIsEncrypted(await cli.getCrypto()?.isEncryptionEnabledInRoom(room.roomId));
+                } catch {
+                    setIsEncrypted(false);
```

Comment:
if `encrypted` is true and the room didn't change but `isEncryptionEnabledInRoom` threw should we really be setting to false? This seems unexpected.

We probably also need to reset to `null` when room changes whilst `func` is calling the async function which could take multiple ticks